### PR TITLE
refactor: avoid extra rerender

### PR DESCRIPTION
**Pull request by hjiog**
## Summary
useAtomValue called rerender inside useEffect, which led to an additional render, and this PR optimized that performance issue.

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

**Review by dai-shi — COMMENTED** [review]
Thanks for the suggestion.
We can't remove `rerender` in useAtomValue, but you are right about the extra rerender (without commits), which isn't desirable.

I'm still working on store2, which will come with a big change, so please be patient.

That said, I think the optimization like what you suggest might be possible and meanwhile what you can help is to create a test case. Thanks for understanding.

File: src/react/useAtomValue.ts [hunk 1]
```diff
@@ -94,7 +94,6 @@ export function useAtomValue<Value>(atom: Atom<Value>, options?: Options) {
       }
       rerender()
     })
-    rerender()
```
Comment (dai-shi) [inline src/react/useAtomValue.ts:97]: This is required. I wonder if we could make a contrived test to cover it.

(I thought there was a rejected PR or something to refer, but can't find it now.)

Comment (hjiog) [reply src/react/useAtomValue.ts:97]: yeah, maybe we should add more test. but the rerender should be optimized.

Comment (dai-shi) [reply src/react/useAtomValue.ts:97]: Did you have a real problem in your usage? If you have, memoization might mitigate the issue for the meantime.

**Comment by dai-shi** [comment]
Please create a new test case in `tests/react/optimization.test.tsx`.
We can't guarantee if it will be solved.

**Comment by gurkerl83** [comment]
> Thanks for the suggestion. We can't remove `rerender` in useAtomValue, but you are right about the extra rerender (without commits), which isn't desirable.
> 
> I'm still working on store2, which will come with a big change, so please be patient.
> 
> That said, I think the optimization like what you suggest might be possible and meanwhile what you can help is to create a test case. Thanks for understanding.

@dai-shi Could you clarify if store2 is intended to fix the underlying cause requiring a secondary render, or have I misunderstood? From your last comment about the test case, it seems this might still be an open question.

Thx!

**Comment by dai-shi** [comment]
> if store2 is intended to fix the underlying cause requiring a secondary render

No, I only mentioned store2 because this PR made changes only to store2.
This behavior/problem isn't specific to store2 or store. It's working as expected.

**Pull request closed without merging**